### PR TITLE
New Resource: `azurerm_mobile_network_sim_group`; New DataSource: `azurerm_mobile_network_sim_group`

### PR DESCRIPTION
**Pull request by ziyeqf**
an overall list is below. Detail of this RP could be found [here](https://learn.microsoft.com/en-us/azure/private-5g-core/)

Though these PRs may have conflicts in merge sequence, but I think it should not block the review progress...

Over view
---
```
order | name 
0     | azurerm_mobile_network                              ✅
1     | azurerm_mobile_network_site                         #20334
1     | azurerm_mobile_network_data_network                 #20338
1     | azurerm_mobile_network_service                      #20337
1     | azurerm_mobile_network_slice                        #20336 
1     | azurerm_mobile_network_sim_group                    <-this one
2     | azurerm_mobile_network_packet_core_control_plane    
2     | azurerm_mobile_network_attached_data_network        
2     | azurerm_mobile_network_sim_policy                   
3     | azurerm_mobile_network_packet_core_data_plane       
4     | azurerm_mobile_network_attached_data_network        
5     | azurerm_mobile_network_sim                           
```

Test
---
```
terraform-provider-azurerm ❯ TF_ACC=1 go test -v ./internal/services/mobilenetwork -run=TestAccMobileNetworkSimGroup -timeout=600m                                                     [ tengzh/mn_sg][$+][🐹 v1.19.5][⏱ 3s][12:32:38]
=== RUN   TestAccMobileNetworkSimGroupDataSource_complete
=== PAUSE TestAccMobileNetworkSimGroupDataSource_complete
=== RUN   TestAccMobileNetworkSimGroup_basic
=== PAUSE TestAccMobileNetworkSimGroup_basic
=== RUN   TestAccMobileNetworkSimGroup_withEncryptionKeyUrl
=== PAUSE TestAccMobileNetworkSimGroup_withEncryptionKeyUrl
=== RUN   TestAccMobileNetworkSimGroup_requiresImport
=== PAUSE TestAccMobileNetworkSimGroup_requiresImport
=== RUN   TestAccMobileNetworkSimGroup_complete
=== PAUSE TestAccMobileNetworkSimGroup_complete
=== RUN   TestAccMobileNetworkSimGroup_update
=== PAUSE TestAccMobileNetworkSimGroup_update
=== CONT  TestAccMobileNetworkSimGroupDataSource_complete
=== CONT  TestAccMobileNetworkSimGroup_requiresImport
=== CONT  TestAccMobileNetworkSimGroup_update
=== CONT  TestAccMobileNetworkSimGroup_complete
=== CONT  TestAccMobileNetworkSimGroup_withEncryptionKeyUrl
=== CONT  TestAccMobileNetworkSimGroup_basic
--- PASS: TestAccMobileNetworkSimGroup_basic (509.04s)
--- PASS: TestAccMobileNetworkSimGroup_requiresImport (527.29s)
--- PASS: TestAccMobileNetworkSimGroupDataSource_complete (985.78s)
--- PASS: TestAccMobileNetworkSimGroup_complete (999.96s)
--- PASS: TestAccMobileNetworkSimGroup_withEncryptionKeyUrl (1003.17s)
--- PASS: TestAccMobileNetworkSimGroup_update (1343.75s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/mobilenetwork 1344.976s

```

### CODE REVIEW

File: website/docs/r/mobile_network_sim_group.html.markdown [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name which should be used for this Mobile Network Sim Groups. Changing this forces a new Mobile Network Sim Group to be created.
+
+* `resource_group_name` - (Required) Specifies the name of the Resource Group where the Mobile Network Sim Groups should exist. Changing this forces a new Mobile Network Sim Group to be created.
```
Comment (katbyte) [inline website/docs/r/mobile_network_sim_group.html.markdown:63]: can this resource belong to a different resource group then the one in mobile_network_id ?

Comment (ziyeqf) [reply website/docs/r/mobile_network_sim_group.html.markdown]: oh, it must be in the same resource group with mobile network. removed it and going to parse from mobile network id.

**Comment by ziyeqf** [comment]
test after update
---
```
terraform-provider-azurerm ❯ TF_ACC=1 go test -v ./internal/services/mobilenetwork -run=TestAccMobileNetworkSimGroup -timeout=600m [ tengzh/mn_sg][$⇡][🐹 v1.19.5][⏱ 0ms][14:31:59]
=== RUN   TestAccMobileNetworkSimGroupDataSource_complete
=== PAUSE TestAccMobileNetworkSimGroupDataSource_complete
=== RUN   TestAccMobileNetworkSimGroup_basic
=== PAUSE TestAccMobileNetworkSimGroup_basic
=== RUN   TestAccMobileNetworkSimGroup_withEncryptionKeyUrl
=== PAUSE TestAccMobileNetworkSimGroup_withEncryptionKeyUrl
=== RUN   TestAccMobileNetworkSimGroup_requiresImport
=== PAUSE TestAccMobileNetworkSimGroup_requiresImport
=== RUN   TestAccMobileNetworkSimGroup_complete
=== PAUSE TestAccMobileNetworkSimGroup_complete
=== RUN   TestAccMobileNetworkSimGroup_update
=== PAUSE TestAccMobileNetworkSimGroup_update
=== CONT  TestAccMobileNetworkSimGroupDataSource_complete
=== CONT  TestAccMobileNetworkSimGroup_requiresImport
=== CONT  TestAccMobileNetworkSimGroup_withEncryptionKeyUrl
=== CONT  TestAccMobileNetworkSimGroup_basic
=== CONT  TestAccMobileNetworkSimGroup_update
=== CONT  TestAccMobileNetworkSimGroup_complete
--- PASS: TestAccMobileNetworkSimGroup_basic (509.20s)
--- PASS: TestAccMobileNetworkSimGroup_requiresImport (530.43s)
--- PASS: TestAccMobileNetworkSimGroupDataSource_complete (727.60s)
--- PASS: TestAccMobileNetworkSimGroup_withEncryptionKeyUrl (756.25s)
--- PASS: TestAccMobileNetworkSimGroup_complete (768.85s)
--- PASS: TestAccMobileNetworkSimGroup_update (969.79s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/mobilenetwork 971.050s                 
```

File: internal/services/mobilenetwork/mobile_network_sim_group_resource.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			properties := resp.Model
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", id)
+			}
+
```
Comment (catriona-m) [inline internal/services/mobilenetwork/mobile_network_sim_group_resource.go:162]:
```suggestion
			if resp.Model == nil {
				return fmt.Errorf("retrieving %s: properties was nil", id)
			}
			properties := *resp.Model

```

File: internal/services/mobilenetwork/mobile_network_sim_group_resource.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+				}
+			}
+
+			properties.SystemData = nil
+
+			if metadata.ResourceData.HasChange("tags") {
+				properties.Tags = &model.Tags
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *properties); err != nil {
```
Comment (catriona-m) [inline internal/services/mobilenetwork/mobile_network_sim_group_resource.go:189]:
```suggestion
			if err := client.CreateOrUpdateThenPoll(ctx, *id, properties); err != nil {
```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for this @ziyeqf, I've left a couple of suggestions inline but in general this is looking good and we can look at merging this soon. Thanks!

**Review by catriona-m — APPROVED** [review]
Thanks @ziyeqf LGTM!

File: internal/services/mobilenetwork/helpers.go [hunk 4]
```diff
@@ -42,3 +43,41 @@ func resourceMobileNetworkChildWaitForDeletion(ctx context.Context, id string, g
 
 	return nil
 }
+
+// some resources defined both systemAssigned and userAssigned Identity type in Swagger but only support userAssigned Identity,
+// so add a workaround to convert type here.
```
Comment (tombuildsstuff) [inline internal/services/mobilenetwork/helpers.go:48]: @ziyeqf is there a Swagger bug for this?

File: internal/services/mobilenetwork/mobile_network_sim_group_resource.go [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+			ValidateFunc: mobilenetwork.ValidateMobileNetworkID,
+		},
+
+		"encryption_key_url": { // needs UserAssignedIdentity
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+		},
+
+		"identity": commonschema.SystemOrUserAssignedIdentityOptional(),
```
Comment (tombuildsstuff) [inline internal/services/mobilenetwork/mobile_network_sim_group_resource.go:67]: the documentation states that only `UserAssigned` is supported.. but this doesn't match - @ziyeqf this'll want updating?

File: internal/services/mobilenetwork/helpers.go (hunk 4 above)
Comment (ziyeqf) [reply internal/services/mobilenetwork/helpers.go:48]: just opened one https://github.com/Azure/azure-rest-api-specs/issues/22634

File: internal/services/mobilenetwork/mobile_network_sim_group_resource.go (hunk 5 above)
Comment (ziyeqf) [reply internal/services/mobilenetwork/mobile_network_sim_group_resource.go:67]: oh, submitted a pr #20474 to update this.

**Pull request merged**